doc/ExpertOptions.md

```diff
@@ -1555,3 +1556,66 @@ dx describe /builds/1.02/applets/hello --json --details | jq '.details | .wdlSou
+
+# Publishing global workflows
+
+<!-- TODO supported version pending release w/ APPS-989 -->
+Publishing a dxCompiler WDL workflow as a global workflow is supported from dxCompiler >= `v2.8.0` and dxpy >= `v<supported version here>`
+
+Example: compiling a WDL workflow for later use as a global workflow.
+```
+java -jar dxCompiler.jar compile <workflow name>.wdl -instanceTypeSelection dynamic
+```
+
+<!-- TODO edit example to use <project id>:<workflow id> once dxpy supports that -->
+Example: publishing a global workflow from a compiled workflow. The global workflow's name will match the WDL workflow name. The global workflow's version must be set with `--version`, since a non-global workflow has no such property. If `--bill-to` is not specified, the publishing user's default billTo will be assumed.
+```
+dx build --globalworkflow --from <workflow id> --version <version> --bill-to <user-xxxx | org-yyyy>
+dx publish globalworkflow-<workflow name>/<version>
+```
+
+Example: adding authorized users to a global workflow
+```
+dx add users globalworkflow-<workflow name> <user-xxxx | org-yyyy | PUBLIC>
+```
+
+See [Limitations](#limitations) below for more details on which parts of the workflow will be automatically included in the global workflow.
+
+## Recommendations
+
+Developers should not store any credentials (passwords, keys, etc.) in the source code of the global workflow, as authorized users will have permission to download and view any applets referenced in the global workflow.
+
+Developers should use simple types for input/output interfaces for global workflows for better platform experience.
+
+For better portability across projects where the workflow will be run, developers should specify runtime resources using memory, disk, CPU requirements (not hard-coded instance types using the key `dx_instance_type`). Developers should compile WDL workflows with `-instanceTypeSelection dynamic`. Expect slightly longer runtimes due to the need to pick instance types at runtime, but this ensures the workflow will always use instance types that are supported for the project where it runs.
+
+For informational purposes, developers should include a reference to a git repo commit containing the original source code. They can add this to the "description" metadata field of the global workflow.
+
+## Limitations
+
+Publishing a dxCompiler workflow as a global workflow is currently only supported for WDL.
+
+The global workflow will currently only support a single region (matching the region in which the original workflow was compiled).
+
+Some dependencies of the original workflow will be automatically included in the global workflow, i.e. they will be cloned into the global workflow's resource container and authorized users of the global workflow will not require additional permissions. These include
+- Applets and sub-workflows that were part of the original workflow
+- Native applets referenced by the workflow
```

Comment:
How are these applets different than applets mentioned in the first point?

Reply:
Edited -- this refers to `dxni` applets, and below, `dxni` apps.